sequencer/finalizer.go

```diff
@@ -281,6 +281,15 @@ func (f *finalizer) newWIPBatch(ctx context.Context) (*WipBatch, error) {
 		return nil, errors.New("state root and local exit root must have value to close batch")
 	}
 
+	// We need to process the batch to update the state root before closing the batch
+	if f.batch.initialStateRoot == f.batch.stateRoot {
```

Comment:
We only need to do this if the batch has no transacgions

Reply:
but if initialStateRoot is equal to the stateRoot this means that the batch is empty (no txs executed), isn't it?